harness/determined/_info.py

```diff
@@ -0,0 +1,326 @@
+import json
+import os
+from typing import Any, Dict, Iterable, List, Optional, Union
+
+from determined import gpu
+
+DEFAULT_RENDEZVOUS_INFO_PATH = "/run/determined/info/rendezvous.json"
+DEFAULT_TRIAL_INFO_PATH = "/run/determined/info/trial.json"
+DEFAULT_RESOURCES_INFO_PATH = "/run/determined/info/resources.json"
+DEFAULT_CLUSTER_INFO_PATH = "/run/determined/info/cluster.json"
+
+
+def getenv_int(key: str) -> Optional[int]:
+    val = os.environ.get(key)
+    return val if val is None else int(val)
+
+
+def as_dict(obj: Any, skip: Iterable[str] = ()) -> Dict[str, Any]:
+    """Remove the leading underscore from private variables for a json representation."""
+    return {k.lstrip("_"): v for k, v in vars(obj).items() if k not in skip}
+
+
+class RendezvousInfo:
+    """
+    RendezvousInfo is machine identity information that is:
+     - configured in the container by the rendezvous layer (when on a Determined cluster)
+     - independent of the launch layer
+     - consumed by the launch layer
+    """
+
+    def __init__(self, container_addrs: List[str], container_rank: int):
+        self.container_addrs = container_addrs
+        self.container_rank = container_rank
+
+    def _to_file(self, path: str = DEFAULT_RENDEZVOUS_INFO_PATH) -> None:
+        """
+        to_file writes the RendezvousInfo to a well-known location in a Determined container.  This
+        is called automatically early in the lifetime of a container, so user code can always expect
+        this file to be written before user code runs.
+        """
+        with open(path, "w") as f:
+            json.dump(vars(self), f)
```

Comment:
It doesn't matter, they would have the same effect.

If `as_dict` did something special or tricky, we could talk about putting it somewhere that it could be accessed from other code, making sure it was uniformly used across the codebase, writing tests for it, etc.  But with how `as_dict` is so simple I'm not even sure having that having it outweighs not having it (DRY principle vs decreasing layers of indirection).